### PR TITLE
FEAT - Unwrap collateral ETH on SL

### PR DESCRIPTION
**Pull request by halaprix**

### CODE REVIEW

File: contracts/commands/AaveStoplLossCommand.sol [hunk 1]
```diff
@@ -297,4 +305,6 @@ contract AaveStoplLossCommand is BaseAAveFlashLoanCommand {
 
         lendingPool.withdraw(collateralTokenAddress, (type(uint256).max), self);
     }
+
+    receive() external payable {}
```
Comment (adamskrodzki) [inline contracts/commands/AaveStoplLossCommand.sol:309]: why ?

Comment (adamskrodzki) [reply contracts/commands/AaveStoplLossCommand.sol:309]: add some validation that at the end there is no eth hanging around

File: contracts/commands/BaseAAveFlashLoanCommand.sol [hunk 2]
```diff
@@ -56,12 +57,14 @@ abstract contract BaseAAveFlashLoanCommand is ICommand, IFlashLoanReceiver {
...
         ILendingPool _lendingPool,
-        AaveProxyActions _aaveProxyActions
+        AaveProxyActions _aaveProxyActions,
+        address _WETH
     ) {
         aaveProxyActions = _aaveProxyActions;
         serviceRegistry = _serviceRegistry;
         lendingPool = _lendingPool;
         self = address(this);
+        WETH = _WETH;
```
Comment (adamskrodzki) [inline contracts/commands/BaseAAveFlashLoanCommand.sol:67]: weth can be added to service registry instead and fetched from there, maybe it is there already

File: contracts/commands/AaveStoplLossCommand.sol (hunk 1 above)
Comment (adamskrodzki) [reply contracts/commands/AaveStoplLossCommand.sol:309]: isExecutionCorrect

**Review by adamskrodzki — CHANGES_REQUESTED** [review]

File: scripts/common/hardhat.utils.ts [hunk 3]
```diff
@@ -339,4 +370,12 @@ export class HardhatUtils {
         const balanceAfter = await token.balanceOf(account)
         return balance == balanceAfter
     }
+    public async replaceImmutableAddress(toReplace: string, replacement: string, contract: Contract) {
+        const bytecode = await this.hre.ethers.provider.getCode(contract.address)
+        const bytecodeWithReplacement = bytecode.replaceAll(
```
Comment (adamskrodzki) [inline scripts/common/hardhat.utils.ts:375]: how do you know that are all ocurancess of an address ?

File: test/aave-stop-loss-multiply.ts [hunk 4]
```diff
@@ -43,9 +44,16 @@ describe('AaveStoplLossCommand-Multiply', async () => {
             ],
         })
         const system = await deploySystem({ utils: hardhatUtils, addCommands: true })
-        const guardDeployerAddress = '0x060c23f67febb04f4b5d5c205633a04005985a94'
+        account = (await hre.ethers.getContractAt(
+            'IAccountImplementation',
+            testPositionAddress,
+        )) as IAccountImplementation
+        const oldGuard = await account.guard()
+        await hardhatUtils.replaceImmutableAddress(hardhatUtils.addresses.DPM_GUARD, oldGuard, system.dpmAdapter!)
```
Comment (adamskrodzki) [inline test/aave-stop-loss-multiply.ts:52]: this is about to brake when hardhatUtils.addresses.DPM_GUARD is changed since blocknumber is fixed

**Review by adamskrodzki — CHANGES_REQUESTED** [review]

File: test/aave-stop-loss-multiply.ts (hunk 4 above)
Comment (halaprix) [reply test/aave-stop-loss-multiply.ts]: It should not. We take the old address of dpm guard - using `account.guard()` and then replace it with current one.

File: scripts/common/hardhat.utils.ts (hunk 3 above)
Comment (halaprix) [reply scripts/common/hardhat.utils.ts:375]: This is only for immutable ones that are stored in runtime bytecode, should work the same as during compilation where al occurances in the code are replaced by the immutable address.

**Pull request merged**